src/libstd/macros.rs

```diff
@@ -64,6 +64,9 @@ macro_rules! panic {
 ///
 /// Equivalent to the `println!` macro except that a newline is not printed at
 /// the end of the message.
+///
+/// Note that because stdout is by default buffered on unix systems, this may
+/// result in output not being emitted until stdout is flushed.
```

Comment:
I think this may want to be rephrased to not mention unix and also mention "line buffering" somewhere, perhaps:

> Note that stdout is frequently line-buffered by default so it may be necessary to use `io::stdout().flush()` to ensure the output is emitted immediately.